### DISCUSSION TITLE
DHCP leases list + Fortitokens list

### DISCUSSION THREAD

**Discussion by rgacz** [FortiPower/PowerFGT#254] [Ideas]
Hello,

Is/could be there a plan to add command

to list current DHCP leases?
Somethning similar to output of
`execute dhcp lease-list`
command?

And to list registered Fortitokens?
Like seen in the grid User & Authentication > Fortitokens?
I guess there is no alternative in CLI for this.

BTW, thank you very much fort PowerFGT module, it helps me with Fortigates analysis a lot. Great job!

**Comment by alagoutte** [comment]
Hi @rgacz 

Good idea for DHCP release !

for waiting, you can use this call API : 

```powershell
(Invoke-FGTRestMethod -uri api/v2/monitor/system/dhcp).results

ip                  : 10.XX.YY.2
reserved            : False
mac                 : 32:df:6e:aa:cc:bb
vci                 : android-dhcp-13
hostname            : Galaxy-A32-5G
expire_time         : 1720552127
status              : leased
interface           : LAN
type                : ipv4
server_mkey         : 1
server_ipam_enabled : False
```

and for FortiToken, you can try this

```powershell
(Invoke-FGTRestMethod -uri api/v2/monitor/user/fortitoken).results
```

**Reply by rgacz** [reply]
Great! Thank you, @alagoutte! 
It took me a while to find out how to enumerate nested PSCustomObjects so that I put it down here for others:
`(Invoke-FGTRestMethod -method "get" -uri "api/v2/monitor/user/fortitoken").results.PSObject.Properties.Value`

**Reply by alagoutte** [reply]
Yes, it is a table with name of serial for header...

can you use also foreach...

**Reply by rgacz** [reply]
Can you give me an example with foreach? It did not work for me.
Same table type is returned by querying URI api/v2/monitor/system/interface.

**Reply by alagoutte** [reply]
a mistake, you need to use PSObject.properties(.value or .name) for enumarate value...

**Comment by alagoutte** [comment]
Added on #276 (DHCP leases) and #279 (Fortitoken)
